backends/verilog/verilog_backend.cc

```diff
@@ -2347,20 +2347,29 @@ void dump_module(std::ostream &f, std::string indent, RTLIL::Module *module)
 	}
 
 	dump_attributes(f, indent, module->attributes, "\n", /*modattr=*/true);
-	f << stringf("%s" "module %s(", indent.c_str(), id(module->name, false).c_str());
-	bool keep_running = true;
+	f << stringf("%s"
+		     "module %s(",
+		     indent.c_str(), id(module->name, false).c_str());
 	int cnt = 0;
-	for (int port_id = 1; keep_running; port_id++) {
-		keep_running = false;
-		for (auto wire : module->wires()) {
-			if (wire->port_id == port_id) {
-				if (port_id != 1)
-					f << stringf(", ");
-				f << stringf("%s", id(wire->name).c_str());
-				keep_running = true;
-				if (cnt==20) { f << stringf("\n"); cnt = 0; } else cnt++;
-				continue;
-			}
+	int max_port_id = 0;
+	for (auto wire : module->wires()) {
```

Comment:
Is pre-allocating the vector with `max_port_id` better than just combining the loops (on line 2355 and 2358) and using `push_back`?

Reply:
Yes it is better, there are holes in the port id sequence, they are not consecutive, the vector is sparsely populated. That way the sorting is happening in o(n)